src/core/utils/request/toPublicUrl.ts

```diff
@@ -9,7 +9,9 @@ export function toPublicUrl(url: string | URL): string {
 
   const urlInstance = url instanceof URL ? url : new URL(url)
 
+  const [, relativeUrl] = urlInstance.href.split(urlInstance.origin)
```

Comment:
This does solve the problem but it changes the other use cases we have for `toPublicUrl`. 

I think a far better approach is to keep these changes scoped to the immediate problem area: the `toPublic()` usage in the `onUnhandledRequest.ts`. 

https://github.com/mswjs/msw/blob/e83a6699bc40a4fc2a593a93346158b8b8b61649/src/core/utils/request/onUnhandledRequest.ts#L24-L27

Let's try something like this:

```diff
const url = new URL(request.url)
const publicUrl = toPublicUrl(url) + url.search
```

This way both relative and absolute URLs will be printed correctly _but_ will also have any search parameters preserved. 

Then, we can modify the tests in `onUnhandledRequest.test.ts` to include two new scenarios:

- Unhandled request with a relative URL and search params;
- Unhandled request with an absolute URL and search params.

Reply:
Thanks for the suggestion. Made changes now and pushed to PR again.